### PR TITLE
modify package.py for SZ to use the latest version v1.4.11

### PR DESCRIPTION
**Pull request by disheng222**
Hi, 
I am the developer of SZ. I updated the package.py for SZ in spack, by using the latest version. 
Please merge it. 
Thanks!

Best,
Sheng

### CODE REVIEW

File: var/spack/repos/builtin/packages/sz/package.py [hunk 1]
```diff
@@ -30,11 +30,11 @@ class Sz(AutotoolsPackage):
     """Error-bounded Lossy Compressor for HPC Data."""
 
     homepage = "https://collab.cels.anl.gov/display/ESR/SZ"
-    url      = "http://www.mcs.anl.gov/~shdi/download/sz-1.4.9-beta.tar.gz"
+    url      = "https://github.com/disheng222/SZ/archive/v1.4.11.0.tar.gz"
 
     version('develop', git='https://github.com/disheng222/SZ.git',
             branch='master')
-    version('1.4.9-beta', '8a00cddc39780c18d766215f69713f82')
```
Comment (davydden) [inline var/spack/repos/builtin/packages/sz/package.py]: we generally keep older ones

**Comment by disheng222** [comment]
Hi Denis,

Do you mean it's better to keep the older version of SZ in Spack?
In fact, the leadership from ECP-CODAR project requested to update it to
the latest version.
Thanks.

Best,
Sheng

2017-09-27 15:20 GMT-05:00 Denis Davydov <notifications@github.com>:

**Comment by davydden** [comment]
> Do you mean it's better to keep the older version of SZ in Spack?

yes. 

p.s. you also need to resolve merge conflicts

**Comment by scheibelp** [comment]
>> Do you mean it's better to keep the older version of SZ in Spack?

> yes.

Since you are the package developer, IMO it can be your choice to keep or leave the version, especially since the initial version is called "beta". In fact it looks like https://github.com/LLNL/spack/pull/5003 already removed the beta version and added `1.4.9.2`

**Comment by disheng222** [comment]
Hi,
Sorry for the misleading.
1.4.11.0 is not a beta version. It should be more stable than 1.4.9.2.

Best,
Sheng

2017-09-27 15:37 GMT-05:00 scheibelp <notifications@github.com>:

**Comment by davydden** [comment]
> it can be your choice to keep or leave the version

not necessarily, it's of course good to remove beta's and subsitute them by the releases, but one shall also keep older releases. the idea is that one should be able to reproduce the stack of software later. So if today someone build with `1.4.11`, this version should be around for some time, even when newer are out.

**Comment by davydden** [comment]
apparently, superseded by https://github.com/LLNL/spack/pull/5517

@disheng222 to be fair, there was no need in creating another PR, you could have resolved the conflict here by rebasing ontop of `develop`, see http://spack.readthedocs.io/en/latest/contribution_guide.html

**Pull request closed without merging**